source/use-cases/sensitive-data-encryption.txt

```diff
@@ -57,12 +57,122 @@ cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Blood Type                  | Random              |  String                                      |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Medical Records             | Random              |  Array                                       |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Insurance: Policy Number    | Deterministic       |  Int (embedded inside `insurance` object)    |
+    +-----------------------------+---------------------+----------------------------------------------+
+    | Insurance: Provider         | Non-Encrypted       |  String (embedded inside `insurance` object) |
+    +-----------------------------+---------------------+----------------------------------------------+
+
+To automatically encrypt and decrypt create a JSON Schema using the information in the data model and pass the schema as a JSON object to the secure client to specify the encrypted field properties.
+In the JSON, you can specify `encryptMetadata <https://docs.mongodb.com/manual/reference/security-client-side-automatic-json-schema/#encryptmetadata-schema-keyword>`_ that is passed down to the fields under the ``properties``
+nested object. 
+
+.. code::
+
+        {
+            "bsonType" : "object",
+            "encryptMetadata" : {
+                "keyId" : // copy and paste your keyID generated here
+            },
+            "properties": {
+                // copy and paste your fields here 
+            }
+        }
+
+Create JSON objects for each field and append them to the `properties` map.
+
+.. code::
+
+    // Deterministic encryption supports querying and is equally as secure as random encryption for high cardinality (very unique) fields such as SSN. A privileged user in the MedcoMD Medical Care Management System might want to query on SSN to verify a patient's identity. One thing to note is that you do not have to include keyId if you specified it in the top-level encryptMetadata field. Though you could include it if you want different users with specific keys to be able to access a specific subset of the encrypted data.
+
+    "ssn": {
+        "encrypt": {
+            "bsonType": "int",
+            "keyId": <keyID>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+        }
+    }
+
+
+.. code:: javascript
+
+    // Random encryption does not support filtering and is best used on fields with low cardinality (very few unique fields) to prevent against frequency analysis attacks. Since there are only 8 blood types, this field is randomly encrypted. To decrypt this data, a privileged user would have to query on another field via a secure collection, and this data would be returned unencrypted.
```

Comment:
The last sentence is unclear to me, especially the part about "via a secure collection". Did you mean to say:
```
This field is decrypted by a CSFLE-enabled client with appropriate access once the document is retrieved.
```

Reply:
I did not mean that, but that's fair so I switched the last sentence to yours.

Reply:
I'm not asking you to write something that you did not mean. What did you mean to write? What does it mean to "query on another field via a secure collection"?

Reply:
I meant that I did not mean those exact words. What I was trying to capture in that last sentence was:
* A privileged user, such as a doctor, can decrypt sensitive data only on deterministically encrypted fields
* A non-privileged user, such as a receptionist, cannot decrypt sensitive data at all

Perhaps a more clear way of stating this would be:

"While a CSFLE-enabled client cannot filter on a randomly encrypted field, it can perform a query on a deterministic field, or a field that is not encrypted at all. Doing so would return the decrypted results of all the fields including randomly encrypted fields, such as blood type."


So the full comment paragraph would read:
```
    // Random encryption does not support filtering and is best used on fields with low cardinality (very few unique fields) to prevent against frequency analysis attacks. Since there are only 8 blood types, this field is randomly encrypted. While a CSFLE-enabled client cannot filter on a randomly encrypted field, it can perform a query on a deterministic field, or a field that is not encrypted at all. Doing so would return the decrypted results of all the fields including randomly encrypted fields, such as blood type.
```

Reply:
What do you think of the updated version ^ @ccho-mongodb

Reply:
It is much more clear. However, there are a couple minor issues:

- First sentence: subject should be "Fields that are randomly encrypted". Otherwise, you are referring to "Random encryption" which does not necessarily have anything to do with MongoDB or filters. Also, it might not be apparent to the reader that you are referring to "query filters".

- Would skip the third and fourth sentences and either replace with a sentence that straightforward such as "While randomly encrypted fields are not queryable, they are decrypted by the client when the document is retrieved."

- There is an extra comma in the second sentence. Corrected: "... on a deterministic field or a field that is not encrypted at all."

- Nitpick: (personal preference and what I was taught) Any quantity less than 10 should be written out; i.e. "eight" instead of 8.

- Nitpick: in written English, we should avoid starting sentences with a gerund if possible ("Doing").